rnd/autogpt_builder/src/app/admin/marketplace/page.tsx

```diff
@@ -2,15 +2,22 @@ import { withRoleAccess } from "@/lib/withRoleAccess";
 
 import React from "react";
 import { getReviewableAgents } from "@/components/admin/marketplace/actions";
-import AdminMarketplaceCard from "@/components/admin/marketplace/AdminMarketplaceCard";
 import AdminMarketplaceAgentList from "@/components/admin/marketplace/AdminMarketplaceAgentList";
+import AdminFeaturedAgentsControl from "@/components/admin/marketplace/AdminFeaturedAgentsControl";
+import { Separator } from "@/components/ui/separator";
 async function AdminMarketplace() {
-  const agents = await getReviewableAgents();
+  const reviewableAgents = await getReviewableAgents();
```

Comment:
You need to useState here to be able to allow the AgentList to update when reviewed:

```
  const [reviewableAgents, setReviewableAgents] = useState([]);

  // Fetch the reviewable agents on component mount
  useEffect(() => {
    async function fetchAgents() {
      const agents = await getReviewableAgents();
      setReviewableAgents(agents);
    }

    fetchAgents();
  }, []);

  // Function to update an individual agent in the list
  const updateAgentInList = (updatedAgent) => {
    setReviewableAgents((prevAgents) =>
      prevAgents.map((agent) =>
        agent.id === updatedAgent.id ? updatedAgent : agent
      )
    );
  };
```

Then pass the update function down to the `AdminMarketplaceCard` and use that when the review is done.

Reply:
done using revalidatePath('/marketplace')